Sources/JS/helpers.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+var maxCollectedProperties = 200
+// groupName: constructor
+var possibleGroups = [
+    String, RegExp, Array, Map, Promise, WeakMap, Set, WeakSet, ArrayBuffer, DataView, Uint8Array, Int8Array,
+    Uint16Array, Int16Array, Uint32Array, Int32Array, Float32Array, Float64Array, Uint8ClampedArray, Object
+]
+// Must be kept in sync with TypeSystem.swift
+baseTypes = {
+    nothing: 0,
+    undefined: 1 << 0,
+    integer: 1 << 1,
+    float: 1 << 2,
+    string: 1 << 3,
+    boolean: 1 << 4,
+    object: 1 << 5,
+    function: 1 << 6,
+    constructor: 1 << 7,
+    unknown: 1 << 8,
+    bigint: 1 << 9,
+    regexp: 1 << 10,
+}
+function setGroup(value, currentType) {
+    for (var i=0;i<possibleGroups.length;i++) {
+        try {
+            if (value instanceof possibleGroups[i]) {
+                currentType.group = possibleGroups[i].name
+                return
+            }
+        } catch(err) {}
+    }
+}
+function isValidPropName(name) {
+    return /^[a-zA-Z_$][0-9a-zA-Z_$]*$/.test(name)
```

Comment:
Maybe link to the specification here to make it clear where this regex comes from?